graspologic/nominate/spectralVN.py

```diff
@@ -0,0 +1,268 @@
+        else:
+            raise NotImplementedError("Multigraph SVN not implemented")
+
+        # Embed graph if embedding not provided
+        if self.embedding is None:
+            if isinstance(self.embedder, BaseSpectralEmbed):
+                self.embedding = self.embedder.fit_transform(X)
+            else:
+                raise TypeError("No embedder available")
+
+    def _predict(self, k: int = 5) -> Tuple[np.ndarray, np.ndarray]:
+        """
+        Nominate vertex based on attribute specific distance from the k nearest neighbors.
+
+        Parameters
+        ----------
+        k : int
+            Number of neighbors to consider in nearest neighbors distance weighting.
+
+        Returns
+        -------
+        An tuple of two np.ndarrays, each with shape = ``(number_vertices, number_attributes_in_seed)``.
+        The array at index 0 is the nomination list. For each attribute column, the rows contain indices
+        of vertices in original adjacency matrix ordered by likelihood of matching that attribute.
+        The array at index 1 is the computed distances, where each element at (i, j) represents the
+        distance metric value between vertex i and attribute j.
+
+        """
+        num_unique = self.unique_att_.shape[0]
+        ordered = self.distance_matrix_.argsort(axis=1)
+        sorted_dists = self.distance_matrix_[np.arange(ordered.shape[0]), ordered.T].T
+        nd_buffer = np.tile(
+            self.attr_labels_[ordered[:, :k]], (num_unique, 1, 1)
+        ).astype(np.float64)
+
+        # comparison taking place in 3-dim view, coordinates produced are therefore 3D
+        inds = np.argwhere(nd_buffer == self.unique_att_[:, np.newaxis, np.newaxis])
+
+        # nans are a neat way to operate on attributes individually
+        nd_buffer[:] = np.NaN
+        nd_buffer[inds[:, 0], inds[:, 1], inds[:, 2]] = sorted_dists[
+            inds[:, 1], inds[:, 2]
+        ]
+
+        # weighting function. Outer inverse for consistency, makes equivalent to simple
+        # ranking by distance in unattributed case, and makes higher ranked vertices
+        # naturally have lower distance metric value.
+        pred_weights = np.power(np.nansum(np.power(nd_buffer, -1), axis=2), -1).T
+
+        nan_inds = np.argwhere(np.isnan(pred_weights))
+        pred_weights[nan_inds[:, 0], nan_inds[:, 1]] = np.inf
+        vert_order = np.argsort(pred_weights, axis=0)
+
+        inds = np.tile(self.unique_att_, (1, vert_order.shape[0])).T
+        inds = np.concatenate((vert_order.reshape(-1, 1), inds), axis=1)
+        pred_weights = pred_weights[inds[:, 0], inds[:, 1]]
+        return vert_order, pred_weights.reshape(vert_order.shape)
+
+    def fit(self, X: np.ndarray, y: np.ndarray):
+        """
+        Constructs the embedding if not provided, then calculates the pairwise distance from each
+        seed to each vertex in graph.
+
+        Parameters
+        ----------
+        X : np.ndarray
```

Comment:
X should be default `None` so if the user doesn't want to pass it, they don't have to type anything

Reply:
Reason I don't do this is that when using sklearn people assume `X` to be the first positional argument (I can't think of any functions where that's not the case?). If I give `X` a default value then `y` will be the first positional argument, which is unexpected. The other option is to also give `y` a default value of `None`, but the algorithm doesn't really make sense in that case. 
I also think that most commonly `X` is provided to fit, so it's not unreasonable to require explicitly setting it to `None` in rarer cases.

Reply:
I think this makes sense, so either an embedding or a adjacency matrix can be provided via `X`, and I'll have an `is_embedding` param as well.

Reply:
keeping it `embedding` is fine too

Reply:
change to `input_graph` for consistency with [LDT](https://graspy.neurodata.io/reference/inference.html#graspologic.inference.LatentDistributionTest)